### ISSUE TITLE
OAuth2ProviderForApplications.GetAuthorizationUrl() NullReferenceException

### ISSUE THREAD

**Issue by jesuissur** [googleads/googleads-dotnet-lib#172]
Hi

We're using the v23.5.0 and got a NullReferenceException when calling the `GetAuthorizationUrl()` method.

Our first debugging session seems to point the problem in the `ParameterUtils.IterateParameters` method.  Shouldn't we have this condition `if (propertyType.GetTypeInfo().IsValueType **&&** value != null)` instead of `if (propertyType.GetTypeInfo().IsValueType **||** value != null)`

This is our code that was working before the update
```
            var config = new AdWordsAppConfig();
            config.OAuth2RedirectUri = GetRedirectUri();

            // Create an OAuth2 object for handling OAuth2 flow.
            var oAuth = new OAuth2ProviderForApplications(config);
            oAuth.IsOffline = true;

            return oAuth.GetAuthorizationUrl();

**Comment by jesuissur** [comment]
Following the quick fix we did in the fork, we got another issue: https://github.com/googleads/googleads-dotnet-lib/issues/173

**Comment by christopherseeley** [comment]
Thanks for reporting this, I'm looking into it now.

**Comment by christopherseeley** [comment]
Fixed in v23.5.1
